crankshaft-config/src/backend/generic/driver.rs

```diff
@@ -15,9 +17,34 @@ pub use shell::Shell;
 /// within a task.
 const DEFAULT_MAX_ATTEMPTS: u32 = 4;
 
+/// The maximum number of attmepts for a driver.
```

Comment:
@peterhuene I forsee doing a lot of this kind of thing, so it deserves some discussion. Essentially, I'd like the configuration objects to represent _exactly_ what was parsed in (so, if the user doesn't provide a configuration for `max_attempts`, the accessor `.max_attempts()` would return `None`), but I'd like for it to be easy to work with the value (e.g., `.unwrap_or_default()`). As such, I plan on doing a decent amount of wrapping these like so.